docs/LoongArch-ELF-ABI-EN.adoc

```diff
@@ -657,3 +657,191 @@ with check 32-bit unsigned overflow
 |Linux, Glibc
 |/lib32/ld-linux-loongarch-ilp32s.so.1
 |===
+
+== Procedure Calling Convention
+
+The basic principle of the LoongArch procedure calling convention is to pass arguments in registers as much as possible (i.e. floating-point arguments are passed in floating-point registers and non floating-point arguments are passed in general-purpose registers, as much as possible); arguments are passed on the stack only when no appropriate register is available.
+
+==== Abbreviations
+
+In this document, *GRLEN* is the bit width of general-purpose register, *FRLEN* is the bit width of floating-point register and *WOA* is the bit width of the argument.
+The argument registers are collectively denoted as *AR* (*GAR* or *FAR*), where the general-purpose argument register is denoted as *GAR* and the floating-point argument register is denoted as *FAR*.
+
+The argument registers are:
+
+. Eight floating-point registers `fa0-fa7` used for passing pass floating-point arguments, and `fa0-fa1` are also used to return values.
+
+. Eight general-purpose registers `a0-a7` used for passing pass integer arguments, with `a0-a1` reused to return values.
+
+Generally, the GARs are used to pass fixed-point arguments, and floating-point arguments when no FAR is available.
+Bit fields are stored in little endian.
+In addition, subroutines should ensure that the values of general-purpose registers `s0-s9` and floating-point registers `fs0-fs7` are preserved across procedure calls.
+
+=== ABI LP64D
+
+That is, *GRLEN* = 64, *FRLEN* = 64.
+
+=== C Data Types and Alignment
+
+The C data types and alignment in the LP64D ABI are defined in the <<Type Size and Alignment, table 3>>.
+
+In most cases, the unsigned integer data types are zero-extended when stored in general-purpose register, and the signed integer data types are sign-extended.
+However, in the *LP64D* ABI, unsigned 32-bit types, such as `*unsigned int*`, are stored in general-purpose registers as proper sign extensions of their 32-bit values.
+
+=== Argument passing
+
+Generally speaking, FARs are only used to pass floating-point arguments, GARs are used to pass non floating-point arguments and floating-point arguments when no FAR is available(long double type is also passed in a pair of GARs) and the reference.
+
+Arguments passed by reference may be modified by the callee.
+
+==== Scalar
+
+There are three cases:
+
+. 0 < WOA ≤ GRLEN
+
+.. Argument is passed in a single AR, or on the stack by value if none is available.
```

Comment:
```suggestion
.. Argument is passed in a single argument register, or on the stack by value if none is available.
```

This is only place where *AR* as abbreviation used. Also expanded version like "argument register" and "argument registers" used a bit in the document too. I would advocate for removal at least this (*AR*) redundant abreviation.

Reply:
Great.